frontend/src/__tests__/cypress/cypress/pages/pipelines/createRunPage.ts

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable camelcase */
+import {
+  PipelineKF,
+  PipelineRunJobKF,
+  PipelineRunKF,
+  PipelineVersionKF,
+  RelationshipKF,
+  ResourceTypeKF,
+} from '~/concepts/pipelines/kfTypes';
+import { buildMockPipelines } from '~/__mocks__/mockPipelinesProxy';
+import { buildMockPipelineVersions } from '~/__mocks__/mockPipelineVersionsProxy';
+
+export class CreateRunPage {
+  protected testId = 'create-run-page';
+
+  find(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.findByTestId(this.testId);
+  }
+
+  findNameInput(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.findByRole('textbox', { name: 'Name' });
+  }
+
+  findDescriptionInput(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return cy.findByRole('textbox', { name: 'Description' });
+  }
+
+  findPipelineSelect(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return this.find().findByTestId('pipeline-toggle-button');
+  }
+
+  findPipelineVersionSelect(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return this.find().findByTestId('pipeline-version-toggle-button');
+  }
+
+  findTriggeredRunTypeRadioInput(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return this.find().findByRole('radio', { name: 'Run once immediately after creation' });
+  }
+
+  findScheduledRunTypeRadioInput(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return this.find().findByRole('radio', { name: 'Schedule recurring run' });
+  }
+
+  findSubmitButton(): Cypress.Chainable<JQuery<HTMLElement>> {
+    return this.find().findByRole('button', { name: 'Create' });
+  }
+
+  fillName(value: string): void {
+    this.findNameInput().type(value);
+  }
+
+  fillDescription(value: string): void {
+    this.findDescriptionInput().type(value);
+  }
+
+  selectPipelineByName(name: string): void {
+    this.findPipelineSelect()
+      .click()
+      .get('[data-id="pipeline-selector-table-list"]')
+      .findByText(name)
+      .click();
+  }
+
+  selectPipelineVersionByName(name: string): void {
+    this.findPipelineVersionSelect()
+      .click()
+      .get('[data-id="pipeline-version-selector-table-list"]')
+      .findByText(name)
+      .click();
+  }
+
+  mockGetPipelines(pipelines: PipelineKF[]): Cypress.Chainable<null> {
+    return cy.intercept(
+      {
+        pathname: '/api/proxy/apis/v1beta1/pipelines',
+      },
+      buildMockPipelines(pipelines),
+    );
+  }
+
+  mockGetPipelineVersions(versions: PipelineVersionKF[]): Cypress.Chainable<null> {
+    return cy.intercept(
+      {
+        method: 'POST',
+        pathname: '/api/proxy/apis/v1beta1/pipeline_versions',
+      },
+      buildMockPipelineVersions(versions),
+    );
+  }
+
+  mockCreateRun(
+    pipelineVersion: PipelineVersionKF,
+    { id, name, description }: Partial<PipelineRunKF>,
+  ): Cypress.Chainable<null> {
+    return cy.intercept(
```

Comment:
any reason not to have these in their own mock function inside __mocks__?

Reply:
The reason is because co-located mock functions with the page objects they're associated with are easier to identify and use, versus searching through a large list of files called "mock", and a list within each file of functions/constants that might work for the page object's purposes, but also might not and have to be adjusted, and those adjustments might impact/break other page objects that might be using the same function, where lots of those mock files/functions have hard-coded material that tests are relying on.

Reply:
Yeah the mocks needs some work, there is a tech debt issue that needs migration: https://github.com/opendatahub-io/odh-dashboard/issues/2264 that addresses this.

Reply:
This is fine though. The testing team will probably sort that tech debt issue into their sprints so we are all on a unified pattern.